implementations/rust/ockam/ockam_api/tests/authority.rs

```diff
@@ -0,0 +1,390 @@
+use ockam::identity::utils::now;
+use ockam::identity::{
+    secure_channels, AttributesEntry, Identifier, SecureChannelOptions, SecureChannels,
+};
+use ockam_api::authenticator::direct::DirectAuthenticatorClient;
+use ockam_api::authority_node::{Authority, Configuration};
+use ockam_api::bootstrapped_identities_store::PreTrustedIdentities;
+use ockam_api::{authority_node, DefaultAddress};
+use ockam_core::{route, Address, Result};
+use ockam_node::{Context, RpcClient};
+use rand::{thread_rng, Rng};
+use std::collections::BTreeMap;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use tempfile::NamedTempFile;
+use tracing::log::warn;
+
+async fn default_configuration() -> Result<Configuration> {
```

Comment:
As a general note, as a reader of tests, I like to see:

 - the tests first where the name reveals the intention or the context
 - the implementation details further down the file (otherwise I need read / analyze structs and functions before even know how / why they will be used)
 - some comments telling me what we are trying to test exactly, what's the rule that we are checking, and what's our strategy for testing it if it is a bit involved

Otherwise our tests serve a non-regression protection but not really as a documentation of the expected behaviour.

Reply:
Fixed